### PR TITLE
feat(cli): migrate `.prettierignore` file

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Follow up PR around the migration from Prettier.

With this PR, Biome reads `.prettierignore` and attempts to add its globs inside the `formatter.ignore` configuration.

I did some tiny refactoring to make the data structure we saved more maintainable and documented. 

## Test Plan

Added more tests

### CODE REVIEW

File: crates/biome_cli/src/execute/migrate/prettier.rs [hunk 1]
```diff
@@ -413,53 +416,84 @@ pub(crate) struct FromPrettierConfiguration {
...
 
-    pub(crate) fn get_biome_configuration(
-        &self,
-    ) -> Option<&(FormatterConfiguration, JavascriptFormatter)> {
-        self.biome_configuration.as_ref()
+    pub(crate) fn store_ignore_path(&mut self, path: impl Into<PathBuf>) {
+        self.ignore_path = Some(path.into())
+    }
+
+    pub(crate) fn as_biome_configuration(&self) -> Configuration {
```
Comment (arendjr) [inline crates/biome_cli/src/execute/migrate/prettier.rs:452]: I think this method should be called `to_biome_configuration()`. Rust has pretty strict conventions around the `as_`, `to_` and `into_` prefixes: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

File: crates/biome_cli/src/execute/migrate/prettier.rs [hunk 2]
```diff
@@ -507,6 +544,33 @@ pub(crate) fn read_prettier_files(
         }
     }
 
+    // to be safe, let's drop this content
+    drop(prettier_config_content);
```
Comment (arendjr) [inline crates/biome_cli/src/execute/migrate/prettier.rs:548]: If you need to do this I’d say it’s a sign you’re doing too much kn a single function.

**Review by arendjr — APPROVED** [review]
Nice!

**Comment by anonrig** [comment]
Can we also migrate `// prettier-ignore` with a simple TODO?

**Review by Conaclos — APPROVED** [review]

**Comment by Sec-ant** [comment]
Hi, just want to ask a question. Is migrating `.prettierignore` an automatic option or an opt-in one? For now I use both Prettier and Biome in my projects because there're still some languages that can only be formatted by Prettier. So I use `.prettierignore` to keep them separated and I don't want Biome to read this file. Is this still possible?

**Comment by arendjr** [comment]
@Sec-ant `.prettierignore` would only be read if you explicitly run `biome migrate prettier`. Your use case should be covered.

**Comment by Sec-ant** [comment]
> @Sec-ant `.prettierignore` would only be read if you explicitly run `biome migrate prettier`. Your use case should be covered.

Great, thanks for the explanation!

**Pull request merged**